server/src/test/java/org/opensearch/indices/replication/SegmentReplicationTargetServiceTests.java

```diff
@@ -127,22 +141,33 @@ public void testAlreadyOnNewCheckpoint() {
         verify(spy, times(0)).startReplication(any(), any(), any());
     }
 
-    public void testShardAlreadyReplicating() {
-        SegmentReplicationTargetService spy = spy(sut);
+    public void testShardAlreadyReplicating() throws InterruptedException {
+        SegmentReplicationTargetService serviceSpy = spy(sut);
```

Comment:
nit - A comment here explaining why we need to spy on the service and what we are asserting would be useful.